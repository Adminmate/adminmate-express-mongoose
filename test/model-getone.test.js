// Hide console.log
global.console = {
  log: jest.fn(),
  // Keep native behaviour for other methods
  error: console.error,
  warn: console.warn,
  info: console.info,
  debug: console.debug,
};

const supertest = require('supertest');
const jwt = require('jwt-simple');

// Include the app
const app = require('./app.js');

// Endpoint prefix
const prefix = '/adminmate/api';

// Generate the admin token
const adminToken = jwt.encode({
  exp_date: Date.now() + 1000
}, 'authkey_secret');

// Before all
beforeAll(done => {
  done();
});

describe('Testing POST /api/models/users/5cd5308e695db945d3cc81a1', () => {
  it('Simple get one route', async () => {
    // Make request
    const response = await supertest(app)
      .post(prefix + '/models/users/5cd5308e695db945d3cc81a1')
      .set('x-access-token', adminToken);

    // Check response
    expect(response.status).toBe(200);
    expect(response.body).toMatchSnapshot();
  });
});

describe('Testing POST /api/models/cars/5cd5308e695db945d3cc81b1', () => {
  it('Simple get one route', async () => {
    // Make request
    const response = await supertest(app)
      .post(prefix + '/models/cars/5cd5308e695db945d3cc81b1')
      .set('x-access-token', adminToken);

    // Check response
    expect(response.status).toBe(200);
    expect(response.body).toMatchSnapshot();
  });
});

// After all
afterAll(done => {
  done();
})