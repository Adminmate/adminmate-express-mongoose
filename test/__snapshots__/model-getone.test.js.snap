// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing POST /api/models/cars/5cd5308e695db945d3cc81b1 Simple get one route 1`] = `
Object {
  "data": Object {
    "_id": "5cd5308e695db945d3cc81b1",
    "createdAt": "2021-04-02T00:00:00.000Z",
    "manufacturer": "Porsche",
    "name": "Porsche 356",
    "updatedAt": "2021-04-02T00:00:00.000Z",
    "user_id": Object {
      "id": "5cd5308e695db945d3cc81a1",
      "label": "5cd5308e695db945d3cc81a1",
      "type": "ref",
    },
    "year": 1948,
  },
  "linkedModels": Array [],
}
`;

exports[`Testing POST /api/models/users/5cd5308e695db945d3cc81a1 Simple get one route 1`] = `
Object {
  "data": Object {
    "_id": "5cd5308e695db945d3cc81a1",
    "birthdate": "2021-04-02T00:00:00.000Z",
    "createdAt": "2021-04-02T00:00:00.000Z",
    "firstname": "John",
    "lastname": "Doe",
    "rating": 4,
    "updatedAt": "2021-04-02T00:00:00.000Z",
  },
  "linkedModels": Array [
    Object {
      "ref_field": "user_id",
      "slug": "cars",
    },
  ],
}
`;
